modules/apps/headless/headless-delivery/headless-delivery-test/src/testFunctional/macros/DocumentFolderAPI.macro

```diff
@@ -58,7 +105,69 @@ definition {
 		}
 	}
 
+	macro assertProperDocumentFolderTotalCount {
+		Variables.assertDefined(parameterList = ${expectedDocumentTotalCount});
+
+		var response = DocumentFolderAPI.getDocumentFoldersWithDifferentPameter(
+			groupName = ${groupName},
+			parameter = ${parameter},
+			parameterValue = ${parameterValue});
+
+		var actualDocumentTotalCount = JSONUtil.getWithJSONPath(${response}, "$..['totalCount']");
+
+		TestUtils.assertEquals(
+			actual = ${actualDocumentTotalCount},
+			expected = ${expectedDocumentTotalCount});
+	}
+
+	macro deleteDocumentFolder {
+		Variables.assertDefined(parameterList = "${documentFolderName},${groupName}");
+
+		var portalURL = JSONCompany.getPortalURL();
+		var response = DocumentFolderAPI.getDocumentFoldersWithDifferentPameter(
+			groupName = ${groupName},
+			parameter = "filter",
+			parameterValue = "name%20eq%20%27${documentFolderName}%27");
+
+		var documentFolderId = JSONPathUtil.getProperty(
+			property = "$..items[0].id",
+			response = ${response});
+
+		var curl = '''
```

Comment:
I think we can use DocumentFolderAPI._curlDocumentFolders() to skip -u&-H codes, right? And what about renaming this method to deleteDocumentFolderByName or ByFolderName?